### PR TITLE
Update prom to one metric per query and Implement extractor

### PR DESCRIPTION
**Pull request by sunya-ch**
This PR includes
- updates on prom module to make query with kepler prefix instead of specific metric name according to https://github.com/sustainable-computing-io/kepler-model-server/issues/74, 
- implementation on default extractor to extract dataframe from prom query according to https://github.com/sustainable-computing-io/kepler-model-server/issues/75.

```py
# tests/extractor_test.py

energy_components = ["package", "core", "uncore", "dram"]
feature_group = 'KubeletOnly'
energy_source = "rapl"

test_extractors = [DefaultExtractor]

...
extractor_instance = extractor()
extracted_data = extractor_instance.extract(query_results, energy_components, \ 
                                             feature_group, energy_source, node_level=True)

```
Example output of DataFrame:
```bash
            kubelet_cpu_usage  kubelet_memory_bytes  package_0_package_power  package_1_package_power  ...  package_0_dram_power  package_1_dram_power  package_2_dram_power  package_3_dram_power
timestamp                                                                                              ...
1675916628             828764            1740500992                  104.807                   74.563  ...                18.349                18.617                17.813                15.836
1675916631             828769            1753092096                  105.148                   74.563  ...                18.384                18.715                17.813                15.948
1675916634             828769            1753157632                  104.807                   74.563  ...                18.349                18.617                17.813                15.836
1675916637             828769            1754656768                   104.94                   74.563  ...                18.349                18.617                17.813                15.836
1675916640             828771            1754771456                  104.807                   74.563  ...                18.349                18.617                17.813                15.836
...                       ...                   ...                      ...                      ...  ...                   ...                   ...                   ...                   ...
1675916913             828956            1871519744                  104.807                   74.563  ...                18.349                18.617                17.813                15.836
1675916916             828956            1871241216                   105.45                   76.134  ...                18.367                19.114                17.846                 16.73
1675916919             828965            1763954688                  105.914                   75.653  ...                18.517                19.024                17.813                 16.14
1675916922             828965            1765707776                  104.807                   74.563  ...                18.349                18.617                17.813                15.836
1675916925             828965            1765728256                  105.625                   74.905  ...                18.349                18.819                17.813                16.092

[100 rows x 18 columns]
```

This PR also includes test and example query output in `tests` folder.

Signed-off-by: Sunyanan Choochotkaew <sunyanan.choochotkaew1@ibm.com>

### CODE REVIEW

File: server/train/train_types.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 CGROUP_FEATURES = ["cgroupfs_cpu_usage_us", "cgroupfs_memory_usage_bytes", "cgroupfs_system_cpu_usage_us", "cgroupfs_user_cpu_usage_us"]
 IO_FEATURES = ["bytes_read", "bytes_writes"]
 BPF_FEATURES = ["cpu_time"]
```
Comment (rootfs) [inline server/train/train_types.py:19]: can you add irq counters too?

Comment (sunya-ch) [reply server/train/train_types.py:19]: I think @KaiyiLiu1234 is working on that in another PR.

Comment (KaiyiLiu1234) [reply server/train/train_types.py:19]: Good thing I waited. I can include change as soon as this PR is merged. The prom queries are very clean now. For the irq counters, I believe we also need to set up the initial model as well.

**Review by KaiyiLiu1234 — COMMENTED** [review]
LGTM. One thing I would like to note is will the prom_output csv files keep increasing in size as query.py is called? The test data is great to use as an example of output but it is also very large.

File: server/train/train_types.py (hunk 1 above)
Comment (KaiyiLiu1234) [reply server/train/train_types.py:19]: I will test it asap

**Comment by sunya-ch** [comment]
> LGTM. One thing I would like to note is will the prom_output csv files keep increasing in size as query.py is called? The test data is great to use as an example of output but it is also very large. Other than that, I can approve immediately if there is nothing else from @rootfs .

No. the prom_output will be updated only when you run `query_test.py` in tests folder to test that prom/query.py works properly.

**Pull request merged**